core/src/consensus.rs

```diff
@@ -1042,46 +1043,83 @@ impl Tower {
         self.last_switch_threshold_check.is_none()
     }
 
-    /// Performs threshold check for `slot`
-    ///
-    /// If it passes the check returns None, otherwise returns Some(fork_stake)
-    pub fn check_vote_stake_threshold(
+    /// Checks a single vote threshold for `slot`
+    fn check_vote_stake_threshold(
+        vote: Option<&Lockout>,
```

Comment:
nit maybe rename to `threshold_vote` so that it's not implied that the vote referenced by `vote_state_before_applying_vote` is this one